### PR TITLE
Fix #1194: emitted warnings that should not be (since `0.51.0`)

### LINKED ISSUE #1194: Emitted warnings that should not be (since `0.51.0`)
# Describe the bug

With the version `0.51.0`, apidoc [emits warnings if there are inconsistencies in the parameters](https://github.com/apidoc/apidoc/pull/1189), but these warnings are also emitted when parameters are present into the query string.

I isolated the origin of the issue here: https://github.com/apidoc/apidoc/blob/a18fd1ee35c540ff171314bf99d393279143a063/lib/core/parser.js#L517-L523

The API parameters are parsed with the URL parts delimited by `/`, but the parse should consider the parts of the query string (e.g. `?foo=:foo&bar=:bar&baz=:baz`.

# Steps to reproduce

`api.js`:
```
/**
 * @api {GET} /my-api?foo=:foo MyApi
 * @apiName MyApi
 * @apiParam {String} foo
 */
```

`logs`:
```log
warn: @apiParam 'foo' was defined but does not appear in URL of @api 'MyApi' in file: 'api.js'
```

# Information

- Version of apidoc: 0.51.0
- Operating system (e.g. GNU+Linux/MacOS/Windows): Windows 10
- Node version: 16.13.2
- Browser (e.g. firefox, chrome): Chrome

### PR DESCRIPTION
**Pull request by BrokenSourceCode**
This PR fixes the issue https://github.com/apidoc/apidoc/issues/1194.

### CODE REVIEW

**Comment by BrokenSourceCode** [comment]
I still have to fix the problem on the template.

File: package.json [hunk 1]
```diff
@@ -56,11 +56,10 @@
     "lodash": "^4.17.21",
     "markdown-it": "^12.2.0",
     "nodemon": "^2.0.15",
-    "path-to-regexp": "^6.2.0",
```
Comment (BrokenSourceCode) [inline package.json:59]: The dependency `path-to-regexp` is now useless.

File: package.json [hunk 2]
```diff
@@ -56,11 +56,10 @@
...
     "prismjs": "^1.25.0",
     "semver": "^7.3.5",
     "style-loader": "^3.3.1",
-    "url-parse": "^1.5.3",
```
Comment (BrokenSourceCode) [inline package.json:63]: The dependency `url-parse` is now useless.

**Comment by BrokenSourceCode** [comment]
I've removed the dependencies [path-to-regexp](https://github.com/pillarjs/path-to-regexp) and [url-parse](https://github.com/unshiftio/url-parse) in favor of [url-polyfill](https://github.com/lifaon74/url-polyfill) (only used for the browser endpoint).

**Comment by NicolasCARPi** [comment]
I have no issues with replacing two dependencies with one, but from what I see on [caniuse](https://caniuse.com/url) we don't even need to care for the polyfill, so this can even be dropped altogether methinks.

**Comment by BrokenSourceCode** [comment]
> I have no issues with replacing two dependencies with one, but from what I see on [caniuse](https://caniuse.com/url) we don't even need to care for the polyfill, so this can even be dropped altogether methinks.

Well, i've removed [url-polyfill](https://github.com/lifaon74/url-polyfill) and a priori the ESLint errors are corrected.

**Pull request closed without merging**